### PR TITLE
Add doc for the current routers and steps to add a new one

### PR DESCRIPTION
**Pull request by vishalya**
## Description
Add the documentation for the routers and how to add a new router.

## Additional context and related issues
NA

## Release notes

(X) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: docs/routers.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+## Routers
```
Comment (mosabua) [inline docs/routers.md:1]: Empty line after each title please .. also start at highest level (#) and dont skip levels

File: docs/routers.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Gateway offers two entry-level router options, providing users with a straightforward and easy-to-use starting point for their routing needs.
```
Comment (mosabua) [inline docs/routers.md:2]: Please wrap all paragraphs at 80 char width

**Review by mosabua — CHANGES_REQUESTED** [review]
Please address first comments .. that will make reviewing easier.

File: docs/routers.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
```
Comment (Chaho12) [inline docs/routers.md:3]: How about adding a title about what info is available for users to use when making a customized module

**Review by Chaho12 — COMMENTED** [review]
oh you added docs yourself. thx :)
fyi. i was waiting for my [pr](https://github.com/trinodb/trino-gateway/pull/398) regarding changes in passing all headers as MultiValueMap to routing manager to be merged.

File: docs/routers.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+To incorporate a new router with advanced capabilities, follow these steps:
+- Derive a class from `RouterBaseModule`
+- This is the module instantiates the router and holds a reference to it.
+- To integrate the new router, you must first add the module name to the
+'modules' section of the configuration file. This tells the gateway to load
+the provider module and make the new router available.
+- e.g. `QueryCountBasedRouterProvider`
+
+### Router class
+- Derive a class from `StochasticRoutingManager`,
```
Comment (mosabua) [inline docs/routers.md:34]:
```suggestion
- Derive a class from `StochasticRoutingManager`
```

File: docs/routers.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+ this is the router which does the actual work
```
Comment (mosabua) [inline docs/routers.md:35]:
```suggestion
 to create the router that does the actual work
```

File: docs/routers.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+- Override the methods `provideAdhocBackend` and `provideBackendForRoutingGroup`
+ for the new smarter logic
```
Comment (mosabua) [inline docs/routers.md:37]:
```suggestion
 and implement the new smarter logic
```

File: docs/routers.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+- The router listens to the list of `ClusterStats`
+via the `updateBackEndStats` method.
+- This method gets called on regular intervals defined
```
Comment (mosabua) [inline docs/routers.md:40]:
```suggestion
- This method is called on regular intervals defined
```

File: docs/routers.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+in the config parameter `monitor=>taskDelaySeconds`
+- Each element in the list corresponds to each backend cluster
+- Only the stats from the healthy cluster are reported, unhealthy clusters
+are not included in the list. so if you have 3 cluster backends and 1 is
```
Comment (mosabua) [inline docs/routers.md:44]:
```suggestion
are not included in the list. If you have three cluster backends and one is
```

File: docs/routers.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+unhealthy then the pamameter `List<ClusterStats> stats` will have only 2 elements.
```
Comment (mosabua) [inline docs/routers.md:45]:
```suggestion
unhealthy, then the parameter `List<ClusterStats> stats` has only two elements.
```

File: docs/routers.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+- In order to get the cluster stats use set the parameter
```
Comment (mosabua) [inline docs/routers.md:46]:
```suggestion
- To get the cluster stats set the parameter
```

File: docs/routers.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+`clusterStatsConfiguration=>monitorType` to `UI_API` or `JDBC`
+which in turn needs the setup of `backendState` section in the config file.
+
+### Config file reference
```
Comment (mosabua) [inline docs/routers.md:50]:
```suggestion
### Configuration file reference

```

File: docs/routers.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+```
```
Comment (mosabua) [inline docs/routers.md:51]:
```suggestion
```yaml
```

**Comment by vishalya** [comment]
I have updated the doc as per the comments, please take a look.

**Review by mosabua — COMMENTED** [review]
Added some final tweaks but this is good to ship. Over time we might need to consolidate some more with the user docs.

**Pull request merged**